### PR TITLE
NAS-107644 / 20.10 / add glustereventsd service api

### PR DESCRIPTION
**Pull request by yocalebo**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/gluster_linux/eventsd.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        `bearer_token` is a bearer token
+        `secret` secret to add JWT bearer token
+        """
+
+        verrors = ValidationErrors()
+
+        # there doesn't seem to be an upper limit on the amount
+        # of webhook endpoints that can be added to the daemon
+        # so place an arbitrary limit of 5 for now
+        if len(self.URLS) == 5:
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/gluster_linux/eventsd.py:87]: What if there are 6?

File: src/middlewared/middlewared/plugins/gluster_linux/eventsd.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            verrors.add(
+                f'webhook_url_create.{data["url"]}',
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/gluster_linux/eventsd.py:89]: Why so weird variable name?

**Review by themylogin — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/gluster_linux/eventsd.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+from middlewared.service import job, accepts, CallError, CRUDService, ValidationErrors
+
+EVENTSD_LOCK = 'gluster_eventsd_lock'
+
+
+class GlusterEventsdService(CRUDService):
+
+    class Config:
+        namespace = 'gluster.eventsd'
+        private = True
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/gluster_linux/eventsd.py:16]: If this is a private API, what/who is going to use it?

**Review by themylogin — APPROVED** [review]
LGTM

**Pull request merged**